### PR TITLE
 Add/Update New Test Screen - conditional questions shown/hidden appropriately

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Notion: https://www.notion.so/joinzoe/COVID-Test-Screen-UI-update-Add-New-Test-Screen-a4fe50cf67eb44b79af00da3ace46652

Note: this is not the complete feature set - the Notion ticket lists out the items still TBD. Wanted to keep PR small as there's quite a lot of logic happening in this one.

**This PR includes the following for the add/update COVID test screens:**
- All questions (conditional on options chosen), along with updated english copy. Swedish and Spanish translations still TBD.
- Update tests screen also dependent on whether it's an old version of the CovidTest or this new version.
- Update version of the Covid test to v2.0.0

**NOTE:**
- We've kept the mechanism field unchanged in the backend. This means we need to convert a user's mechanism choice to the "old version" for saving.
  - If lateral flow chosen ⇒ saved as `nose_and_throat_swab` + `is_rapid_test: true`
  - If PCR chosen ⇒ saved as `nose_and_throat_swab` + `is_rapid_test: false`
- Updating of old tests
    - Mechanism will be shown as one of new tests, whatever they logged in the past:
        - If nose/throat swab/spit + rapid test ⇒ Lateral flow
        - If nose/throat swab/spit + not rapid test ⇒ PCR
        - If blood_sample ⇒ **assumed finger prick**
        - If finger_price ⇒ finger prick
        - If needle_draw ⇒ needle draw
        - If other ⇒ other
    - User is allowed to amend `location` of test , `invited_to_test`, and `is_rapid_test` if these were filled out previously
    - The version of the test does NOT change even if they update it
- Changed question from "Did a trained worked swab you?" to "Who performed the test?" which has different answers. Updating an old test therefore leaves this question-answer blank.
- Similarly the question-answer will be blank for "Was it Anti-N or Anti-S" which is a new question.
- Swedish users
    - Have removed option for `NOSE_OR_THROAT_SWAB_AND_SALIVA` when adding new tests

**Won't fix**
- There's some extra spacing between title of Add new test screen and "* required field" text. This is because both our Header component and FormWrapper components have inbuilt padding. This is where we need to figure out how to make design and our components consistent so will leave this for now.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Jul-22-2021 23-41-03

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [x] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: assets/lang/en.json [hunk 1]
```diff
@@ -284,14 +284,24 @@
     "picker-nose-throat-swab": "A swab of my nose or throat",
     "picker-nose-throat-swab-and-saliva": "A swab of my nose or throat and saliva",
     "picker-saliva-sample": "I spat in a cup/tube",
+    "picker-lateral": "Lateral flow test (Antigen)",
+    "picker-pcr": "PCR test (Antigen)",
     "picker-blood-sample": "I gave a blood sample",
-    "picker-finger-prick": "A finger-prick blood test",
-    "picker-blood-draw": "A blood test, done using a needle",
+    "picker-finger-prick": "A finger-prick blood test (Antibody)",
+    "picker-blood-draw": "A blood test, done using a needle \n(Antibody)",
```
Comment (aikewoody) [inline assets/lang/en.json:291]: We shouldn't be using newlines in strings, there are so many different screen sizes that there is a high chance on certain resolutions it will look weird

File: src/core/user/dto/CovidTestContracts.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-
 export type CovidTest = {
   id: string; // Test schema version
   version: string; // Test schema version
   patient: string;
   type: CovidTestType;
 
   result: string;
-  mechanism: CovidTestMechanismOptions;
+  mechanism: string;
```
Comment (aikewoody) [inline src/core/user/dto/CovidTestContracts.ts:8]: Is this undo intentional?

File: src/core/user/dto/UserAPIContracts.ts [hunk 3]
```diff
@@ -49,8 +51,16 @@ export enum CovidTestMechanismOptions {
 
 export enum CovidTestTrainedWorkerOptions {
   TRAINED = 'trained',
-  UNTRAINED = 'untrained',
-  UNSURE = 'unsure',
+  SELF_NO_SUPERVISION = 'self_no_supervision',
+  SELF_SUPERVISION = 'self_supervision',
+  UNTRAINED = 'untrained', // Deprecated
+  UNSURE = 'unsure', // Deprecated
```
Comment (aikewoody) [inline src/core/user/dto/UserAPIContracts.ts:57]: Can these deprecated options be deleted easily?

File: src/features/covid-tests/helpers/index.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  CovidTestMechanismOptions.BLOOD_NEEDLE_DRAW,
+];
+
+const ALL_MECHANISMS_EXCEPT_OTHER = PCR_LATERAL_FLOW_MECHANISMS.concat(ANTIBODY_MECHANISMS);
+
+const ZOE_INVITE_MECHANISMS = [CovidTestMechanismOptions.BLOOD_FINGER_PRICK, CovidTestMechanismOptions.PCR];
+
+const ALL_MECHANISMS = ALL_MECHANISMS_EXCEPT_OTHER.concat(ZOE_INVITE_MECHANISMS);
+
+export {
```
Comment (aikewoody) [inline src/features/covid-tests/helpers/index.ts:23]: Wouldn't it be less code by directly exporting the consts? Btw I love this helper file! Makes the codebase much cleaner

**Review by aikewoody — APPROVED** [review]
Made a few comments about small things but otherwise looks good!

File: src/core/user/dto/CovidTestContracts.ts (hunk 2 above)
Comment (juliatan) [reply src/core/user/dto/CovidTestContracts.ts:8]: Yep - it was originally string which I changed to be stricter in my previous branch... then realised why they had originally set to string.

File: src/core/user/dto/UserAPIContracts.ts (hunk 3 above)
Comment (juliatan) [reply src/core/user/dto/UserAPIContracts.ts:57]: The backend is still TBD so have left it in for now.

Comment (selvach) [reply src/core/user/dto/UserAPIContracts.ts:57]: These value might be returned by the backend when you list out / get COVID Tests.  A user may create the COVID Test with an older screen version (e.g. last year) and then then go in and edit that COVID Test. 

So there are few edge cases to take care of... :(

File: assets/lang/en.json (hunk 1 above)
Comment (juliatan) [reply assets/lang/en.json:291]: Ah yeh, forgot to mention this in the notes. I was trying to work out why this is happening. The bottom component which uses the same radio button seems to be fine. Any ideas what might be causing this. Seems strange to have to box just this question in for padding...

Screenshot 2021-07-23 at 09 43 07

File: src/features/covid-tests/helpers/index.ts (hunk 4 above)
Comment (juliatan) [reply src/features/covid-tests/helpers/index.ts:23]: Not sure what you mean - sent a message in Slack to clarify

File: assets/lang/en.json (hunk 1 above)
Comment (aikewoody) [reply assets/lang/en.json:291]: I see, this can probably be fixed by setting flex on the Text component

File: src/core/user/dto/CovidTestContracts.ts (hunk 2 above)
Comment (aikewoody) [reply src/core/user/dto/CovidTestContracts.ts:8]: Got it, I liked your initial change by making it more strict but can understand there is a specific reason this is not possible in the end

File: src/core/user/dto/UserAPIContracts.ts (hunk 3 above)
Comment (aikewoody) [reply src/core/user/dto/UserAPIContracts.ts:57]: Ah got it, maybe good to put this short explanation in the comments
